src/renderer.zig

```diff
@@ -611,6 +617,222 @@ pub const MenuBackGround = struct {
+			}
+
+			pos = vec.normalize(pos)*@as(Vec3f, @splat(starDist));
+
+			const normPos = vec.normalize(pos);
+
+			const col = getStarColor(temperature*5772.0, light, starColorImage);
+
+			const lat: f32 = @floatCast(std.math.asin(normPos[2]));
+			const lon: f32 = @floatCast(std.math.atan2(-normPos[0], normPos[1]));
+
+			const mat = Mat4f.rotationZ(lon).mul(Mat4f.rotationX(lat));
+
+			const posA = vec.xyz(mat.mulVec(.{triVertA[0], triVertA[1], triVertA[2], 1.0}));
+			const posB = vec.xyz(mat.mulVec(.{triVertB[0], triVertB[1], triVertB[2], 1.0}));
+			const posC = vec.xyz(mat.mulVec(.{triVertC[0], triVertC[1], triVertC[2], 1.0}));
+
+			starData[i*20 ..][0..3].* = posA;
+			starData[i*20 + 4 ..][0..3].* = posB;
+			starData[i*20 + 8 ..][0..3].* = posC;
+
+			starData[i*20 + 12 ..][0..3].* = pos;
+			starData[i*20 + 16 ..][0..3].* = col;
+		}
+
+		starSsbo = graphics.SSBO.initStatic(f32, &starData);
+
+		c.glGenVertexArrays(1, &starVao);
+		c.glBindVertexArray(starVao);
+		c.glEnableVertexAttribArray(0);
+
+		skyShader = Shader.initAndGetUniforms("assets/cubyz/shaders/skybox/sky.vs", "assets/cubyz/shaders/skybox/sky.fs", "", &skyUniforms);
+		skyShader.bind();
+
+		const rawData = [_]f32{
+			-1, -1, -1,
+			1,  -1, -1,
+			1,  1,  -1,
+			-1, 1,  -1,
+			-1, -1, 1,
+			1,  -1, 1,
+			1,  1,  1,
+			-1, 1,  1,
+		};
+
+		const indices = [_]c_int{
+			0, 3, 1, 1, 3, 2,
+			5, 6, 4, 4, 6, 7,
+			3, 7, 2, 2, 7, 6,
+			1, 5, 0, 0, 5, 4,
+			4, 7, 0, 0, 7, 3,
+			1, 2, 5, 5, 2, 6,
+		};
+
+		c.glGenVertexArrays(1, &skyVao);
+		c.glBindVertexArray(skyVao);
+		c.glGenBuffers(2, &skyVbos);
+		c.glBindBuffer(c.GL_ARRAY_BUFFER, skyVbos[0]);
+		c.glBufferData(c.GL_ARRAY_BUFFER, @intCast(rawData.len*@sizeOf(f32)), &rawData, c.GL_STATIC_DRAW);
+		c.glVertexAttribPointer(0, 3, c.GL_FLOAT, c.GL_FALSE, 3*@sizeOf(f32), null);
+		c.glEnableVertexAttribArray(0);
+		c.glBindBuffer(c.GL_ELEMENT_ARRAY_BUFFER, skyVbos[1]);
+		c.glBufferData(c.GL_ELEMENT_ARRAY_BUFFER, @intCast(indices.len*@sizeOf(c_int)), &indices, c.GL_STATIC_DRAW);
+	}
+
+	pub fn deinit() void {
+		starShader.deinit();
+		starSsbo.deinit();
+		c.glDeleteVertexArrays(1, &starVao);
+
+		skyShader.deinit();
+		c.glDeleteVertexArrays(1, &skyVao);
+		c.glDeleteBuffers(2, @ptrCast(&skyVbos));
+	}
+
+	pub fn render() void {
+		c.glDisable(c.GL_CULL_FACE);
+		c.glDisable(c.GL_DEPTH_TEST);
+
+		const viewMatrix = game.camera.viewMatrix;
+		skyShader.bind();
+
+		const time = game.world.?.gameTime.load(.monotonic);
+
+		var starOpacity: f32 = 0;
+		const dayTime = @abs(@mod(time, game.World.dayCycle) -% game.World.dayCycle/2);
+		if(dayTime < game.World.dayCycle/4 - game.World.dayCycle/16) {
+			starOpacity = 1;
+		} else if(dayTime > game.World.dayCycle/4 + game.World.dayCycle/16) {
+			starOpacity = 0;
+		} else {
+			starOpacity = @as(f32, @floatFromInt(dayTime - (game.World.dayCycle/4 - game.World.dayCycle/16)))/@as(f32, @floatFromInt(game.World.dayCycle/8));
```

Comment:
This logic is inverted at dawn, causing the stars to suddenly disappear, then slowly reappear, then disappear again.